playbooks/deployments.md

```diff
@@ -0,0 +1,137 @@
+
+### Approaches:
+
+#### Run a single command from a developer's check-out
+
+E.g.:
+
+- `hokusai pipeline promote` (Metaphysics, Diffusion, Vibrations...)
+- `git push heroku master` (...?)
+
+This approach is simple and a straightforward usage of our existing tools, BUT:
+
+- Requires a working environment (at least with `hokusai`/`git` and credentials)
+- May have a different effect depending on the local state (e.g., of git remotes or `hokusai/*.yml` configurations)
+- Isn't adaptable to multi-step deploys (e.g., compiling assets)
+- Doesn't enforce the exact deploy commands (though they may be documented)
+- May not match the code and configuration used (by CI) to deploy to staging
+
+#### Run a jenkins job
+
+E.g.:
+
+- [deploy-gravity-production](https://joe.artsy.net/job/deploy-gravity-production/)
+- [deploy-gemini-production](https://joe.artsy.net/job/deploy-gemini-production/)
+
+Conveniently, jenkins jobs are visible to others. They leverage a precisely configured (and thus repeatable)
+environment, and even prevent tasks from overlapping. However:
+
+- Spinning up Jenkins slaves simply to execute a deploy command feels like overkill
+- Once again, the set-up for such a job may not match that used by CI to deploy to staging
+
+#### Open and merge a PR to trigger a release
+
+E.g.:
+
+- via `hokusai pipeline promote` ([Volt](https://github.com/artsy/volt/blob/master/.circleci/config.yml), ...)
+- via `hokusai production deploy ...`
+  ([Kaws](https://github.com/artsy/kaws/blob/2e8e0ca8be2bc14e30e979cf2481e01767762f5e/.circleci/config.yml),
+  [Force](https://github.com/artsy/force/blob/4e81b8f92d40bcdcd6c575be31ac561e1500f203/.circleci/config.yml))
+- via `git push heroku master` ([Impulse](https://github.com/artsy/impulse/blob/master/.circleci/config.yml))
+
+This leverages the exact set-up already in use to deploy staging, and is independent of any developer's
+environment. The pull request also acts as a convenient record of the deploy, including commits and any discussion.
+It's possible, however, for deploys to end up blocked or delayed by other CI tasks. This approach also requires a
+little extra CircleCI configuration.
+
+Some projects trigger a deploy by PR-ing `master` to a `release` branch. This risks including commits that haven't
+completed a full CI run over the `master` branch. To avoid this, others update a `staging` or `master-stable`
+branch after the staging deploy is complete and PR from that branch.
+
+## Recommendations
+
+- It's acceptable for projects to start by recommending a basic `hokusai pipeline promote`
+- When setting up projects, use `hokusai setup` in combination with the templates in
+  [artsy/artsy-hokusai-templates](https://github.com/artsy/artsy-hokusai-templates). They provide reasonable
+  starting points for `.circleci/config.yml` and other configuration files.
+- Full-fledged projects should drive releases (via CircleCI) with pull requests from a `staging` branch to a
```

Comment:
There's been an issue with people having chrome extensions that delete git branches that cause them to accidentally delete `staging` if they run a deploy. I wonder if it's worth mentioning here (maybe not).

Reply:
Tell me more...

It's common for repos to have long-running branches be the base of PRs (e.g., a `develop` branch in some flows), so such an extension would seem to be very dangerous.

Reply:
i.e. https://artsy.slack.com/archives/C0C4AJ1PF/p1499794598511441